src/Framework/Constraint/IsResourceOfType.php

```diff
@@ -9,48 +9,47 @@
  */
 namespace PHPUnit\Framework\Constraint;
 
-use PHPUnit\Util\InvalidArgumentHelper;
-
 /**
- * Constraint that checks if value is resource type
- *
- * The file path to check is passed as $other in evaluate().
+ * Constraint that checks if value is a resource of given type
  */
-class IsResourceType extends Constraint
```

Comment:
removal of `IsResourceType` is BC breaker that will break a lot of custom Constraints.
hint: leave `IsResourceType` as is and introduce `IsResourceOfType` as a new class